third_party/arcs/examples/VisionPipeline.arcs

```diff
@@ -0,0 +1,58 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+
+// This is supposed to be about the same as the example policy in 
+// src/analysis/souffle/tests/taint_with_logic/taint_with_auth_logic.dl
+// particularly under the banner "Program Manifests"
+
+particle ImageDetector
+    id_sensor_data_packet: reads SensorDataPacket {
+        camera_feed: Number,
+        video_resolution: Number
+    }
+    id_image_detection_model: reads ImageDetectionModel {}
+    id_detection_boxes: writes ImageDetectionBoxes {}
+        // the label of id_sensor_packet.video_resolution is the empty set (public)
+    claim id_sensor_data_packet.camera_feed is raw_video_tag
+    claim id_image_detection_model is image_detection_model_tag
+        // id_detection_boxes is downgraded from
```

Comment:
We could also remove a label by using `claim x is not tag` syntax.  Although, Mark may not be supporting it yet.

Reply:
`claim x is not tag` seems to be about the same as downgrading, so if our intention is to use authorization logic for downgrading, then it seems like having this second way of downgrading would create an opportunity to subvert the policy specified by the authorization logic.

Part of the idea is also to only allow the owner of the data to decide where the data can be moved and how it can be used (though sometimes this could happen indirectly through delegation). The rule for downgrading checks that the owner of the data is doing the downgrade, whereas the `claim x is not tag` syntax does not have a speaker, so I think we can't ensure that only the owners of data can set policies for their data.

If we eventually want to prove hyperproperties that prevent active attacks (i.e. ones like robust declassification that show only Bob can influence release of Bob's data), having a principal attributed to causing the downgrade seems important. (I think the principal that `says` the downgrade should happen can be used in the way that integrity would normally be used).

I think the ideal syntax would also be closer to `speakerPrincipal says downgrades(handle_name, old_label, new_label)` where the important difference is that because there is a `new_label` field, this can be used to both remove tags and add new ones -- this maps really well to the common pattern where data is transformed into a "redacted" version of the data (e.g. `ReviewGroup says downgrades(handle_name, precise_location, approximate_location)`. (`ReviewGroup` can then go on to specify different data usage policies for `precise_location` and `approximate_location`). (The only reason why this demo is not closer to this ideal syntax is because we are currently doing the full analysis in datalog, and that both makes things trickier to figure out and has some expressiveness limitations).